Sources/Basics/HTTPClient/HTTPClientRequest.swift

```diff
@@ -0,0 +1,105 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the Swift open source project
+//
+// Copyright (c) 2020-2023 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See http://swift.org/LICENSE.txt for license information
+// See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+
+// `AsyncFileSystem` used by `HTTPClientRequest` requires Swift Concurrency back-deployment.
+#if swift(>=5.5.2)
+
+import Foundation
+
+import struct TSCBasic.AbsolutePath
+import protocol TSCBasic.FileSystem
+
+public struct HTTPClientRequest: Sendable {
```

Comment:
This type couldn't be reused in both clients, thus it's dual to `LegacyHTTPClientRequest`, but has no references to Dispatch types, such as `DispatchQueue`.